src/middleware/protected-route.ts

```diff
@@ -0,0 +1,15 @@
+import { NextFunction, Request, Response } from "express";
+
+export const protectedRoute = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  if (req.user) {
+    next();
+  } else {
+    return res
+      .status(401)
+      .send({ error: "Unauthorized access to protected route" });
+  }
+};
```

Comment:
This is middleware we can all use in the routes we define to prevent the route from being accessed by users who have no valid session 😀